ambry-cloud/src/main/java/com/github/ambry/cloud/azure/ADAuthBasedStorageClient.java

```diff
@@ -67,6 +75,17 @@ public ADAuthBasedStorageClient(BlobServiceClient blobServiceClient, BlobBatchCl
     super(blobServiceClient, blobBatchClient, azureMetrics, blobLayoutStrategy);
   }
 
+  /**
+   * Note that this method is not thread safe. Its need to be called from a thread safe context.
```

Comment:
This method is always called from a thread safe context. The reason for the comment is to avoid thread unsafe usage in future. The only place `buildBlobServiceClient` is called from is `StorageClient::createBlobStorageClient`.

`createBlobStorageClient` itself is called either from `StorageClient`'s constructor or from `ADAuthBasedStorageClient::refreshStorageToken`, both of them are thread safe contexts.